src/group_theory/submonoid/membership.lean

```diff
@@ -233,12 +234,19 @@ lemma pow_right_injective_iff_pow_injective {n : M} :
   function.injective (λ m : ℕ, n ^ m) ↔ function.injective (pow n) :=
 subtype.coe_injective.of_comp_iff (pow n)
 
-theorem log_pow_eq_self [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)) (m : ℕ) :
-  log (pow n m) = m :=
-pow_right_injective_iff_pow_injective.mp h $ pow_log_eq_self _
+/-- The exponentiation map is an isomorphism from the additive monoid on natural numbers to powers
+when it is injective. The inverse is given by the logarithms. -/
+@[simps]
+def pow_log_equiv [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)):
```

Comment:
I wonder if we should switch `h` to `order_of n ≠ 0` at some point; I think it's out of scope for this PR, anyways.

Reply:
Should be a separate theorem or do you suggest changing `function.injective` to `order_of n = 0`? I'm looking at the definition of `order_of` and it's `noncomputable`.

Reply:
ooh, I didn't realize but makes sense; we should probably keep it this way, then. It would definitely be nice to have the fact that if the order is zero then the power is injective; not sure if we have that.